### PR TITLE
Addition of field in form schema and correcting the test cases accordingly

### PR DESCRIPTION
**Pull request by denduluriramasaimohanakhil**
dev:
  lint:0
  coverage:85%
filterChanges:
  lint:0
  coverage:85%

Added pushToElastic field in schema

### CODE REVIEW

File: src/main/java/com/techsophy/tsf/runtime/form/changelog/Elastic.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+@Slf4j
+@RequiredArgsConstructor
+public class Elastic
+{
+    private  final MongoTemplate mongo;
+
+    @Execution
+    public void setElasticPushField()
+    {
+        Query query = new Query();
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/changelog/Elastic.java]: this should be protected and only happen if the elastic enabled environment is set

File: src/test/java/com/techsophy/tsf/runtime/form/changelog/AddGroupTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import java.io.InputStream;
+import java.util.Collections;
+
+import static com.techsophy.tsf.runtime.form.constants.FormModelerConstants.TP_FORM_DEFINITION_COLLECTION;
+import static com.techsophy.tsf.runtime.form.constants.FormModelerConstants.UNDERSCORE_ID;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+
+@ExtendWith(MockitoExtension.class)
+class AddGroupTest
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/changelog/AddGroupTest.java:25]: why are these tests being added?

File: src/main/java/com/techsophy/tsf/runtime/form/changelog/Elastic.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+import static com.techsophy.tsf.runtime.form.constants.FormModelerConstants.EXECUTION_IS_FAILED;
+
+@ChangeUnit(id="push-to-elastic",order = "5")
+@Slf4j
+@RequiredArgsConstructor
+public class Elastic
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/changelog/Elastic.java]: Give a proper name

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java [hunk 4]
```diff
@@ -184,7 +185,7 @@ void saveFormDataWithoutCollectionTest() throws IOException
                 TEST_VERSION,testFormData,testFormMetaData);
         doReturn(userList).when(mockUserDetails).getUserDetails();
         FormResponseSchema formResponseSchemaTest = new FormResponseSchema(TEST_FORM_ID, TEST_NAME, TEST_COMPONENTS, list,TEST_PROPERTIES,TEST_TYPE_FORM, TEST_VERSION,IS_DEFAULT_VALUE, TEST_CREATED_BY_ID,
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java]: replace this constructor

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledExceptionTest.java [hunk 5]
```diff
@@ -124,7 +125,7 @@ void saveFormDataWithoutCollectionInvalidInputExceptionTest() throws IOException
                 TEST_VERSION,testFormData,testFormMetaData);
         doReturn(userList).when(mockUserDetails).getUserDetails();
         FormResponseSchema formResponseSchemaTest = new FormResponseSchema(TEST_FORM_ID, TEST_NAME, TEST_COMPONENTS, list,TEST_PROPERTIES,TEST_TYPE_FORM, TEST_VERSION,IS_DEFAULT_VALUE, TEST_CREATED_BY_ID,
-                String.valueOf(TEST_CREATED_ON), TEST_UPDATED_BY_ID, String.valueOf(TEST_UPDATED_ON));
+                String.valueOf(TEST_CREATED_ON), TEST_UPDATED_BY_ID, String.valueOf(TEST_UPDATED_ON), Status.DISABLED);
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledExceptionTest.java]: this need not be provided to the constructor

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java [hunk 6]
```diff
@@ -250,7 +251,7 @@ void saveFormDataWithUniqueKeyCheckCollectionNotExist() throws IOException
...
-                String.valueOf(TEST_CREATED_ON), TEST_UPDATED_BY_ID, String.valueOf(TEST_UPDATED_ON));
+                String.valueOf(TEST_CREATED_ON), TEST_UPDATED_BY_ID, String.valueOf(TEST_UPDATED_ON),Status.DISABLED);
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java]: refactor to not user constructor

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java [hunk 7]
```diff
@@ -420,6 +421,9 @@ void saveFormDataEmptyDocumentIdTest() throws IOException
         Mockito.when(mockObjectMapper.convertValue(any(),eq(FormDataDefinition.class))).thenReturn(new FormDataDefinition());
         Document document=new Document();
         document.append("version",1);
+        FormResponseSchema formResponseSchemaTest = new FormResponseSchema(TEST_FORM_ID, TEST_NAME, TEST_COMPONENTS, list,TEST_PROPERTIES,TEST_TYPE_FORM, TEST_VERSION,IS_DEFAULT_VALUE, TEST_CREATED_BY_ID,
+                String.valueOf(TEST_CREATED_ON), TEST_UPDATED_BY_ID,String.valueOf(TEST_UPDATED_ON),Status.DISABLED);
+        Mockito.when(mockFormService.getRuntimeFormById(anyString())).thenReturn(formResponseSchemaTest);
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java:178]: refactor to replace the constructor

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

File: src/main/java/com/techsophy/tsf/runtime/form/changelog/ElasticFieldMigration.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+@RequiredArgsConstructor
+public class ElasticFieldMigration
+{
+    private  final MongoTemplate mongo;
+
+    @Execution
+    public void setElasticPushField()
+    {
+        Query query = new Query();
+        Update update = new Update().set("elasticPush", Status.ENABLED);
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/changelog/ElasticFieldMigration.java]: this value needs to be set as per the environment variable value

File: src/test/java/com/techsophy/tsf/runtime/form/changelog/AddGroupTest.java (hunk 2 above)
Comment (denduluriramasaimohanakhil) [reply src/test/java/com/techsophy/tsf/runtime/form/changelog/AddGroupTest.java:25]: To improve the code coverage to above 85% I have added above test cases extra

File: src/main/java/com/techsophy/tsf/runtime/form/changelog/Elastic.java (hunk 1 above)
Comment (denduluriramasaimohanakhil) [reply src/main/java/com/techsophy/tsf/runtime/form/changelog/Elastic.java]: Ok modifying it accordingly now

File: src/main/java/com/techsophy/tsf/runtime/form/changelog/Elastic.java (hunk 3 above)
Comment (denduluriramasaimohanakhil) [reply src/main/java/com/techsophy/tsf/runtime/form/changelog/Elastic.java]: Yes changed the class name now

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java (hunk 4 above)
Comment (denduluriramasaimohanakhil) [reply src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java]: Yes Replacing with No argument constructor

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledExceptionTest.java (hunk 5 above)
Comment (denduluriramasaimohanakhil) [reply src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledExceptionTest.java]: Correcting as above

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java (hunk 6 above)
Comment (denduluriramasaimohanakhil) [reply src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java]: Yes doing the same as above

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java (hunk 7 above)
Comment (denduluriramasaimohanakhil) [reply src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java:178]: Yes doing the same as above

File: src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataElasticAspectHandler.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+@Aspect
+@Component
+@RequiredArgsConstructor
+public class FormDataElasticAspectHandler
+{
+    private final FormService formService;
+    private final ObjectMapper objectMapper;
+    private final FormDataElasticService formDataElasticService;
+
+    @Before("execution(* com.techsophy.tsf.runtime.form.controller.FormDataController..*(..))")
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataElasticAspectHandler.java]: just explain if we need before or after, what would match the current behavior

File: src/main/java/com/techsophy/tsf/runtime/form/entity/Auditable.java [hunk 10]
```diff
@@ -23,4 +14,6 @@
     private String updatedById;
     @LastModifiedDate
     private String updatedOn;
+    @Version
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/entity/Auditable.java]: dont use audit version, lets set created attributes manually

File: src/main/java/com/techsophy/tsf/runtime/form/changelog/ElasticFieldMigration.java [hunk 11]
```diff
@@ -7,24 +7,32 @@
...
+
+import static com.techsophy.tsf.runtime.form.constants.FormModelerConstants.ELASTIC_ENABLE;
 import static com.techsophy.tsf.runtime.form.constants.FormModelerConstants.EXECUTION_IS_FAILED;
 
 @ChangeUnit(id="push-to-elastic",order = "5")
 @Slf4j
 @RequiredArgsConstructor
 public class ElasticFieldMigration
 {
+    @Value(ELASTIC_ENABLE)
```
Comment (VaibhavTechsophy) [inline src/main/java/com/techsophy/tsf/runtime/form/changelog/ElasticFieldMigration.java]: pls dont add the annotations via constants

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

File: src/main/java/com/techsophy/tsf/runtime/form/entity/Auditable.java (hunk 10 above)
Comment (denduluriramasaimohanakhil) [reply src/main/java/com/techsophy/tsf/runtime/form/entity/Auditable.java]: Ok sir will go manually only

File: src/main/java/com/techsophy/tsf/runtime/form/changelog/ElasticFieldMigration.java (hunk 8 above)
Comment (denduluriramasaimohanakhil) [reply src/main/java/com/techsophy/tsf/runtime/form/changelog/ElasticFieldMigration.java]: ok vaibhav modifying it as per env value

File: src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataElasticAspectHandler.java (hunk 9 above)
Comment (denduluriramasaimohanakhil) [reply src/main/java/com/techsophy/tsf/runtime/form/aspect/FormDataElasticAspectHandler.java]: I will explain vaibhav

File: src/main/java/com/techsophy/tsf/runtime/form/changelog/ElasticFieldMigration.java (hunk 11 above)
Comment (denduluriramasaimohanakhil) [reply src/main/java/com/techsophy/tsf/runtime/form/changelog/ElasticFieldMigration.java]: Yes replacing with string only

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataElasticServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    private final String gatewayURL="http://localhost:8080";
+    private final boolean elasticSource=true;
+
+    @Test
+    void saveOrUpdateToElasticTest()
+    {
+        FormDataElasticServiceImpl mockFormDataElasticServiceImpl=
+                new FormDataElasticServiceImpl(mockWebClientWrapper,mockTokenUtils,gatewayURL,elasticSource);
+        FormDataDefinition formDataDefinition=new FormDataDefinition();
+        Assertions.assertDoesNotThrow(()->mockFormDataElasticServiceImpl.saveOrUpdateToElastic(formDataDefinition));
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/runtime/form/service/FormDataElasticServiceTest.java]: this is not an appropriate test case with mocked data. You should verify that the downstream is called with expected inputs at least.

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

File: src/test/java/com/techsophy/tsf/runtime/form/service/FormDataElasticServiceTest.java (hunk 12 above)
Comment (denduluriramasaimohanakhil) [reply src/test/java/com/techsophy/tsf/runtime/form/service/FormDataElasticServiceTest.java]: corrected this testcase

**Review by VaibhavTechsophy — APPROVED** [review]

**Pull request merged**